backend/api/quivr_api/modules/sync/utils/dropboxutils.py

```diff
@@ -0,0 +1,308 @@
+import os
+import uuid
+from datetime import datetime, timedelta, timezone
+from io import BytesIO
+
+import dropbox
+from fastapi import UploadFile
+from pydantic import BaseModel, ConfigDict
+from quivr_api.logger import get_logger
+from quivr_api.modules.brain.repository.brains_vectors import BrainsVectors
+from quivr_api.modules.knowledge.repository.storage import Storage
+from quivr_api.modules.sync.dto.inputs import (
+    SyncFileInput,
+    SyncFileUpdateInput,
+    SyncsActiveUpdateInput,
+)
+from quivr_api.modules.sync.repository.sync_files import SyncFiles
+from quivr_api.modules.sync.service.sync_service import SyncService, SyncUserService
+from quivr_api.modules.sync.utils.list_files import (
+    get_dropbox_files_by_id,
+    list_dropbox_files,
+)
+from quivr_api.modules.sync.utils.upload import upload_file
+from quivr_api.modules.upload.service.upload_file import check_file_exists
+
+logger = get_logger(__name__)
+
+APP_KEY = os.getenv("DROPBOX_APP_KEY")
+APP_SECRET = os.getenv("DROPBOW_CONSUMER_SECRET")
+
+
+class DropboxSyncUtils(BaseModel):
+    model_config = ConfigDict(arbitrary_types_allowed=True)
+
+    storage: Storage
+    sync_files_repo: SyncFiles
+    sync_active_service: SyncService
+    sync_user_service: SyncUserService
+
+    async def _upload_files(
+        self,
+        token_data: dict,
+        files: list,
+        current_user: str,
+        brain_id: str,
+        sync_active_id: int,
+    ):
+        """
+        Download files from DropBox.
+
+        Args:
+            credentials (dict): The credentials for accessing Google Drive.
+            files (list): The list of file metadata to download.
+
+        Returns:
+            dict: A dictionary containing the status of the download or an error message.
+        """
+        dbx = dropbox.Dropbox(token_data["access_token"])
+        dbx.check_and_refresh_access_token()
+        token_data["access_token"] = dbx._oauth2_access_token
+
+        downloaded_files = []
+        bulk_id = uuid.uuid4()
+        for file in files:
+            logger.info("🔥🔥🔥🔥: %s", file)
```

Comment:
I hate to do remove fire emojies 😢  but we likely need to parse or search logs so a more structured logging is needed.